definitions/tools/merge_bams.cwl

```diff
@@ -3,26 +3,81 @@
 cwlVersion: v1.0
 class: CommandLineTool
 label: "Sambamba: merge"
-baseCommand: ["/usr/bin/sambamba_merge"]
+baseCommand: ["/bin/bash", "merge.sh"]
 requirements:
     - class: ResourceRequirement
       ramMin: 8000
       coresMin: 4
     - class: DockerRequirement
-      dockerPull: "mgibio/biscuit:0.3.8.2"
+      dockerPull: "mgibio/bam-merge:0.1"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'merge.sh'
+        entry: |
+            #!/bin/bash
+            set -o pipefail
+            set -o errexit
+            set -o nounset
+
+            while getopts "t:m:n:s:" opt; do
+                case "$opt" in
+                    t)
+                        NTHREADS="$OPTARG"
+                        ;;
+                    m)
+                        MEM="$OPTARG"
+                        ;;
+                    n)
+                        OUTFILENAME="$OPTARG"
+                        ;;
+                    s)
+                        SORTED="$OPTARG"
+                        ;;
+                esac
+            done
+            BAMS=${@:$OPTIND}
+            NUM_BAMS=`echo "$# - $OPTIND + 1" | perl -nae 'print eval $_'` #can't use typical dollar/parens bash math
+
+            #if there is only one bam, just copy it and index it
+            if [[ $NUM_BAMS -eq 1 ]]; then
+                cp "$BAMS" "$OUTFILENAME";
+            else
+                if [[ $SORTED == "true" ]];then
+                    /usr/bin/sambamba merge -t "$NTHREADS" "$OUTFILENAME" "$BAMS"
+                else #unsorted bams, use picard
+                    MEM=`echo "$MEM/1000" | perl -nae 'print eval $_'`
+                    cmd="java -jar -Xmx"
+                    cmd+="$MEM"
```

Comment:
I don't believe it's safe to give all of the memory to java's heap.  But even if it were, it might be better to use a unit signifier letter to avoid the math.  (Like `"-Xmx" . $MEM . "m"` if it were a perl expression).